src/envoy/http/path_rewrite/filter.cc

```diff
@@ -105,6 +105,12 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool) {
   }
 
   std::string new_path;
+
+  // This route config mismatch shouldn't happen as the current route match is
+  // regenerated and ordered in the control planeafter syntax parsing uri
+  // template in the same logic of path rewrite filter , to ensure the request
+  // is specifically matched to targeted uri template according to its grammar.
+  // It would be a bug in the route match generation if t happens.
```

Comment:
Just say:  It should be a bug in Envoy RouteMatch generated by config-mgr if url_template is mismatched with the request path

Reply:
Done.